### PR TITLE
Fix PE-D bugs

### LINKED ISSUE #178: [PE-D] Absentees window is not automatically closed
How to reproduce: Open the absentees window by clicking `absentees`, then close the application;
Expected: The window will be closed as well.
Actual: It is not closed

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: zunedz/ped#5

### PR DESCRIPTION
**Pull request by EvaderFati**
Closes #178 #184 #186 #196 #200 #201

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 1]
```diff
@@ -5,12 +5,20 @@
  */
 public class Messages {
 
+    public static final String MESSAGE_DUPLICATE_CLASS_GROUP = "This class already exists in TAssist!\n"
+            + "Classes of the same module are identified by their CLASS_GROUP_IDs, which are case-insensitive";
```
Comment (lkldev) [inline src/main/java/seedu/address/commons/core/Messages.java:9]: should be class group id and module index

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 2]
```diff
@@ -5,12 +5,20 @@
...
+    public static final String MESSAGE_DUPLICATE_MODULE = "This module already exists in TAssist!\n"
+            + "Modules in the same academic year are identified by their MODULE_CODEs, which are case-insensitive";
```
Comment (lkldev) [inline src/main/java/seedu/address/commons/core/Messages.java:11]: should module code and academic year

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 3]
```diff
@@ -5,12 +5,20 @@
...
+    public static final String MESSAGE_DUPLICATE_STUDENT = "This student already exists in TAssist!\n"
+            + "Students are identified by their STUDENT_IDs, which are case-insensitive";
```
Comment (lkldev) [inline src/main/java/seedu/address/commons/core/Messages.java:13]: missing assessment, which are identified by either
assessment name + module index OR
simple name + module index

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 4]
```diff
@@ -91,7 +92,16 @@ public CommandResult execute(Model model) throws CommandException {
...
-            throw new CommandException(MESSAGE_DUPLICATE_ENTITY); // TODO: Update Command Exception
+            switch (toAdd.getEntityType()) {
+            case STUDENT:
+                throw new CommandException(Messages.MESSAGE_DUPLICATE_STUDENT);
+            case TA_MODULE:
+                throw new CommandException(Messages.MESSAGE_DUPLICATE_MODULE);
+            case CLASS_GROUP:
+                throw new CommandException(Messages.MESSAGE_DUPLICATE_CLASS_GROUP);
+            default:
+                throw new CommandException(Messages.MESSAGE_UNKNOWN_ENTITY);
```
Comment (lkldev) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:103]: missing case assessment

**Review by lkldev — COMMENTED** [review]
missing assessment when checking for entity, and a few comments about duplicate entities

**Review by lkldev — APPROVED** [review]

**Review by wxliong — APPROVED** [review]
LGTM! :)

**Pull request merged**